File: lib/models/image_slide/image_slide_parameter.dart
class ImageSlideParameter {
  String readableName;
  String value;
  String type;

  ImageSlideParameter({
    required this.readableName,
    required this.value,
    required this.type,
  });

  Map<String, dynamic> toMap() {
    return {'readable_name': readableName, 'value': value, 'type': type};
  }

  factory ImageSlideParameter.fromJson(Map<String, dynamic> json) {
    return ImageSlideParameter(
      readableName: json['readable_name'],
      value: json['value'],
      type: json['type'],
    );
  }

  ImageSlideParameter copyWith({
    String? readableName,
    String? value,
    String? type,
  }) {
    return ImageSlideParameter(
      readableName: readableName ?? this.readableName,
      value: value ?? this.value,
      type: type ?? this.type,
    );
  }
}

File: python-app/models/image_slide/image_slide_parameter.py
class ImageSlideParameter():
    def __init__(self, value: str, readable_name: str, _type: str):
        self.value = value
        self.readable_name = readable_name
        self.type = _type
    
    def to_dict(self) -> dict[str, any]:
        return {
            'value': self.value,
            'readable_name': self.readable_name,
            'type': self.type
        }
